src/blocks/data-block/TableBlock.tsx

```
@@ -1,58 +1,269 @@
+      return data;
+    }
+
+    return data.filter(row => {
+      return filterGroups.every(group => {
+        const selectedOptionValue = activeFilters[group.id];
+        if (!selectedOptionValue || selectedOptionValue === ALL_FILTER_VALUE) {
+          return true; // No filter applied for this group or 'All' is selected
+        }
+
+        const selectedOption = group.options.find(opt => opt.value === selectedOptionValue);
+        if (selectedOption && selectedOption.filterFn) {
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)